graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/user/TopSecondDegreeByCountForUser.java

```diff
@@ -47,28 +47,53 @@ public TopSecondDegreeByCountForUser(
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
 
-  @Override
-  protected boolean isEdgeEngagementWithinAgeLimit(
+  /**
+   * Return whether the edge is within the age limit which is specified in the request.
+   * It is used to filter information of unwanted edges from being aggregated.
+   * @param request       is the request given by the requester
+   * @param edgeIterator  is the iterator being used to iterate node's edges.
+   *                      It carries information such as the engagement time of the current edge
+   * @return true if this edge is within the max age limit, false otherwise.
+   */
+  private boolean isEdgeEngagementWithinAgeLimit(
       TopSecondDegreeByCountRequestForUser request,
       EdgeIterator edgeIterator) {
     long keepEdgeWithinTime = request.getMaxEdgeEngagementAgeInMillis();
     long edgeEngagementTime = ((TimestampEdgeIterator)edgeIterator).getCurrentEdgeEngagementTimeInMillis();
     return (edgeEngagementTime >= System.currentTimeMillis() - keepEdgeWithinTime);
   }
+  /**
+   * Only social proof types specified in the user request are counted
+   * For example, a request's social proof types only contain "Follow", and a node has "Follow" and "Mention" edges.
+   * Only the "Follow" edge will be counted, and the "Mention" edge is considered invalid
+   */
+  private boolean isEdgeTypeValid(byte[] validEdgeTypes, byte edgeType) {
```

Comment:
There is a similar type of filtering (but not exactly the same functionality) provided by SocialProofTypesFilter. I am not sure if this filter is used in the ResultFilterChain for the User request, but it's worth checking. 

For the Tweet request, we don't remove any socialProofs from the NodeInfo, but we check that the candidate contains at least one of the social proofs contained in request.getSocialProofTypes(). I don't know what the desired behavior is.